apps/web/src/features/walletconnect/services/WalletConnectWallet.ts

```diff
@@ -25,25 +25,47 @@ function assertWeb3Wallet<T extends Web3WalletType | null>(web3Wallet: T): asser
  */
 class WalletConnectWallet {
   private web3Wallet: Web3WalletType | null = null
+  private initPromise: Promise<Web3WalletType> | null = null
 
   /**
    * Initialize WalletConnect wallet SDK
    */
   public async init() {
     if (this.web3Wallet) return
 
-    const core = new Core({
-      projectId: WC_PROJECT_ID,
-      logger: IS_PRODUCTION ? undefined : 'debug',
-      customStoragePrefix: LS_NAMESPACE,
-    })
+    if (!this.initPromise) {
+      this.initPromise = (async () => {
+        const core = new Core({
+          projectId: WC_PROJECT_ID,
+          logger: IS_PRODUCTION ? undefined : 'debug',
+          // Global Core keys:
+          // - This implementation: globalThis._walletConnectCore_SAFE_v2__wc_dapp_
+          // - @web3-onboard: globalThis._walletConnectCore_ (empty suffix, default)
+          //
+          // This isolation ensures wallet connections work independently from dApp connections.
+          customStoragePrefix: LS_NAMESPACE + 'wc_dapp_',
```

Comment:
As I understand it, this is the actual fix. The rest of the PR just converts async-await to a then-catch promise that seems logically equivalent and unnecessary.